File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        vector<vector<int>> res;

        for(int i = 0 ; i < n-2 ; i++) {

            if (i > 0 && nums[i] == nums[i - 1]) continue;
            
            int low = i+1, high = n-1;

            while(low < high) {
                int sum = nums[i] + nums[low] + nums[high];

                if(sum == 0){
                    res.push_back({nums[i], nums[low], nums[high]});
                    low++;
                    high--;

                    while(nums[low] == nums[low-1]) low++;
                    while(nums[high] == nums[high+1]) high--;
                }
                else if(sum < 0) low++;
                else high--;
            }
        }

        return res;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}
        freq = [[] for i in range(len(nums)+1)]

        for n in nums:
            mp[n] = 1 + mp.get(n, 0);

        for n, c in mp.items():
            freq[c].append(n)
        
        res = []

        for i in range(len(freq)-1, 0, -1):
            for n in freq[i]:
                res.append(n)
                if len(res) == k:
                    return res
        
        return []